docs/drafts/how-to-z-stream-upgrades.md

```diff
@@ -0,0 +1,23 @@
+## How-to: Z-Stream Automatic Updates
+
+To restrict automatic updates to only z-stream patches and avoid breaking changes, use the `"~"` version range operator when setting the version for the desired package in Catalog source.
+
+Example:
+
+```yaml
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  installNamespace: argocd
+  source:
+    sourceType: Catalog
+    catalog:
+      packageName: argocd-operator
+      version: “~2.3" # Automatically upgrade patch releases for v2.3
```

Comment:
is `2.3.x` equivalent to this?

Should we document all the semantic ways to achieve "z-stream automatic updates"?

Reply:
I don't think we need to document the synonyms for a particular version range. I think one example is sufficient here. The API reference will document all the ways.

Reply:
Yeah, that was my thought process as well. I think it's best to give folks one way of doing things (even if there are multiple - and document that elsewhere). Options lead to stress XD - having said that, is 2.3.x prettier?

Reply:
i think `~2.3` is fine. it's 1 character shorter :P